content.go

```diff
@@ -42,3 +43,34 @@ func Tag(ctx context.Context, target Target, src, dst string) error {
 	}
 	return target.Tag(ctx, desc, dst)
 }
+
+var (
+	// DefaultResolveOptions provides the default ResolveOptions.
+	DefaultResolveOptions = ResolveOptions{
+		TargetPlatform: nil,
+	}
+)
+
+// ResolveOptions contains parameters for oras.Resolve.
+type ResolveOptions struct {
+	// TargetPlatform is the target platform.
+	// Will do the platform selection if specified.
+	TargetPlatform *ocispec.Platform
+}
+
+// Resolve resolves a descriptor with provided reference from the target.
+func Resolve(ctx context.Context, target Target, ref string, opts ResolveOptions) (ocispec.Descriptor, error) {
+	desc, err := target.Resolve(ctx, ref)
```

Comment:
@Wwwsylvia @shizhMSFT If 1) The target is an ReferenceFetcher and 2) the input ref is a tag, do we need to use FetchReference to save 1 http request?